src/routers/WormholeL2VoteAggregatorRouter.sol

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import {L2VoteAggregator} from "src/L2VoteAggregator.sol";
+import {WormholeL2VoteAggregator} from "src/WormholeL2VoteAggregator.sol";
+
+contract WormholeL2VoteAggregatorRouter is WormholeL2VoteAggregator {
+  /// @dev Thrown when calldata is invalid for the provided function ID.
+  error InvalidCalldata();
+
+  /// @dev Thrown when calldata provides a function ID that does not exist.
+  error FunctionDoesNotExist();
+
+  /// @dev Thrown when a function is not supported.
+  error UnsupportedFunction();
+
+  /// @param _votingToken The address of the L2 token used for voting.
+  /// @param _relayer The address of the Wormhole relayer contract.
+  /// @param _governorMetadata The address of the L2 Governor Metadata contract which holds proposal
+  /// state.
+  /// @param _l1BlockAddress The address of the contract used to fetch the L1 block number.
+  /// @param _sourceChain The Wormhole chain ID of the source chain when sending messages.
+  /// @param _targetChain The Wormhole chain ID of the target chain when sending messages.
+  constructor(
+    address _votingToken,
+    address _relayer,
+    address _governorMetadata,
+    address _l1BlockAddress,
+    uint16 _sourceChain,
+    uint16 _targetChain
+  )
+    WormholeL2VoteAggregator(
+      _votingToken,
+      _relayer,
+      _governorMetadata,
+      _l1BlockAddress,
+      _sourceChain,
+      _targetChain
+    )
+  {}
+
+  /// @dev if we remove this function solc will give a missing-receive-ether warning because we have
+  /// a payable fallback function. We cannot change the fallback function to a receive function
+  /// because receive does not have access to msg.data. In order to prevent a missing-receive-ether
+  /// warning we add a receive function and revert.
+  receive() external payable {
+    revert UnsupportedFunction();
+  }
+
+  /// @dev Wrap the default `castVote` method in a method that optimizes the calldata.
+  /// @param _msgData Optimized calldata for the `castVote` method. We restrict proposalId to a
+  /// `uint16` rather than the default `uint256`.
+  function _castVote(bytes calldata _msgData) internal {
```

Comment:
The main optimizations are:

1. Using packed data
2. Using a fallback function rather than a named function
3. Restricting the proposalId to a uint16

Reply:
https://github.com/ScopeLift/l2-flexible-voting/pull/69/commits/8a1d28a543332220b02e6c6521d1a2000ac03375